homeassistant/components/light/avion.py

```diff
@@ -42,24 +43,17 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     lights = []
     if CONF_USERNAME in config and CONF_PASSWORD in config:
-        data = avion.avion_info(config[CONF_USERNAME], config[CONF_PASSWORD])
-        for location in data['locations']:
-            for avion_device in location['location']['devices']:
-                device = {}
-                mac = avion_device['device']['mac_address']
-                device['name'] = avion_device['device']['name']
-                device['key'] = location['location']['passphrase']
-                device['address'] = '%s%s:%s%s:%s%s:%s%s:%s%s:%s%s' % \
-                                    (mac[8], mac[9], mac[10], mac[11], mac[4],
-                                     mac[5], mac[6], mac[7], mac[0], mac[1],
-                                     mac[2], mac[3])
-                lights.append(AvionLight(device))
+        devices = avion.get_devices(
+            config[CONF_USERNAME], config[CONF_PASSWORD])
```

Comment:
Use `.get()`. No need to define a default in the configuration validation for the username and the password as `None` is the [default](https://docs.python.org/3.8/library/stdtypes.html#dict.get) already. 

Thus I don't agree with [this comment](https://github.com/home-assistant/home-assistant/pull/17780#pullrequestreview-168204198). Sometimes you want the `KeyError` to happen.

Reply:
This is inside a conditional block which ensures those keys exist, so I don't think `.get()` is necessary.

```python
     if CONF_USERNAME in config and CONF_PASSWORD in config:
         devices = avion.get_devices(
             config[CONF_USERNAME], config[CONF_PASSWORD])
```